docs/docs/code/python/delay/README.md

```diff
@@ -0,0 +1,122 @@
+
+[[toc]]
+
+## Using `pd.flow.delay`
+
+`pd.flow.delay` takes one argument: the number of **milliseconds** you'd like to pause your workflow until the next step executes. {{$site.themeConfig.DELAY_MIN_MAX_TIME}}.
+
+Note that [delays happen at the end of the step where they're called](#when-delays-happen).
+
+```python
+import random
+
+def handler(pd: 'pipedream'):
+    # Delay a workflow for 60 seconds (60,000 ms)
+    pd.flow.delay(60 * 1000)
+
+    # Delay a workflow for 15 minutes
+    pd.flow.delay(15 * 60 * 1000)
+
+    # Delay a workflow based on the value of incoming event data,
+    # or default to 60 seconds if that variable is undefined
+    pd.flow.delay(pd.steps["trigger"].get("event", {}).get("body", {}).get("delayMs", 60 * 1000))
+
+    # Delay a workflow a random amount of time
+    pd.flow.delay(random.randint(0, 999))
+```
+
+::: tip Paused workflow state
+
+When `pd.flow.delay` is executed in a Python step, the workflow itself will enter a **Paused** state.
+
+While the workflow is paused, it will not incur any credits towards compute time. You can also [view all paused workflows in the Event History](/event-history/#filtering-by-status).
+
+:::
+
+### Credit usage
+
+The length of time a workflow is delayed from `pd.flow.delay` does _not_ impact your credit usage. For example, delaying a 256 megabyte workflow for five minutes will **not** incur ten credits.
+
+However, using `pd.flow.delay` in a workflow will incur two credits.
+
+One credit is used to initially start the workflow, then the second credit is used when the workflow resumes after its pause period has ended.
+
+## `cancel_url` and `resume_url`
+
+Both the built-in **Delay** actions and `pd.flow.delay` return a `cancel_url` and `resume_url` that lets you cancel or resume paused executions.
+
+These URLs are specific to a single execution of your workflow. While the workflow is paused, you can load these in your browser or send an HTTP request to either:
+
+- Hitting the `cancel_url` will immediately cancel that execution
+- Hitting the `resume_url` will immediately resume that execution early
+
+[Since Pipedream pauses your workflow at the _end_ of the step where you run call `pd.flow.delay`](#when-delays-happen), you can send these URLs to third party systems, via email, or anywhere else you'd like to control the execution of your workflow.
+
+```python
+import requests
+
+def handler(pd: 'pipedream'):
+  links = pd.flow.delay(15 * 60 * 1000)
+  # links contains a dictionary with two entries: resume_url and cancel_url
+
+  # Send the URLs to a system you own
+  requests.post("https://example.com", json=links)
+
+  # Email yourself the URLs. Click on the links to cancel / resume
+  pd.send.email(
+    subject=f"Workflow execution {pd.steps['trigger']['context']['id']}",
+    text=f"Cancel your workflow here: {links['cancel_url']} . Resume early here: {links['resume_url']}",
+    html=None
```

Comment:
Note: In Python's `pd.send.email`, `html` defaults to `""`, so it overrides `text` in the email unless it (`html`) is explicitly set to `None`.

Reply:
That's really good to know, might want to consider that as a `::: warning` in this doc below this code example!

Reply:
Added the `::: warning` in https://github.com/PipedreamHQ/pipedream/pull/10192/commits/2ee07e1f4a50b01fe26b5d6083baecaa1e1a5620